onnxruntime/core/mlas/lib/qladd_avx2.cpp

```diff
@@ -0,0 +1,195 @@
+#include "mlasi.h"
+
+template <typename DataType>
+static
+MLAS_FORCEINLINE
+__m256i
+MlasShiftRight24Epi32(__m256i v);
+
+template <> __m256i MlasShiftRight24Epi32<int8_t>(__m256i v) { return _mm256_srai_epi32(v, 24); }
+template <> __m256i MlasShiftRight24Epi32<uint8_t>(__m256i v) { return _mm256_srli_epi32(v, 24); }
+
+template <typename DataType>
+MLAS_FORCEINLINE
+static
+__m256i
+MlasPackS16_256(__m256i a, __m256i b);
+
+template <> __m256i MlasPackS16_256<uint8_t>(__m256i a, __m256i b) { return _mm256_packus_epi16(a, b); }
+template <> __m256i MlasPackS16_256<int8_t>(__m256i a, __m256i b) { return _mm256_packs_epi16(a, b); }
+
+template<typename DataType, bool IsScalarA, bool IsScalarB>
+void
+MlasQLinearAddKernelAvx2Helper(
+    const DataType* InputA,
+    float ScaleA,
+    int32_t ZeroPointA,
+    const DataType* InputB,
+    float ScaleB,
+    int32_t ZeroPointB,
+    float ScaleC,
+    int32_t ZeroPointC,
+    DataType* OutputC,
+    size_t N
+    )
+{
+    const float ScaleRatio_AC = ScaleA / ScaleC;
+    const float ScaleRatio_BC = ScaleB / ScaleC;
+    const __m256 VectorScaleRatio_AC = _mm256_set1_ps(ScaleRatio_AC);
```

Comment:
>_mm256_set1_ps [](start = 39, length = 14)

Do you need to add avx/avx2 option to call it?

Reply:
Totally for avx2+fma. Aarch controlled in camke file.